File: 1/nim/problem_two.nim
from strutils import split, strip
from taximap import TaxiMap, Intersection, newTaxiMap, step, blocks_away, repeated_points

var
  f: File

if open(f, "../input"):
  var
    map: TaxiMap = newTaxiMap()
    repeats: seq[Intersection]

  let line = readLine(f)
  let instructions = split(line, ',')
  for instruction in instructions:
    map.step(strip(instruction))
  repeats = map.repeated_points()
  echo "Blocks Away: ", repeats[0].blocks
else:
  echo "Couldn't open input file."

File: 1/python/problem_two.py
#!/usr/bin/env python

from taximap import TaxiMap


def main():
    taxi_map = TaxiMap()
    with open('../input', 'rb') as handle:
        for instruction in map(str.strip, handle.read().split(',')):
            taxi_map.step(instruction)
        first_repeat = taxi_map.repeated_points()[0]
        print "Blocks Away:", first_repeat.blocks

if __name__ == '__main__':
    main()
